### PR TITLE
Update example config file for uberbank workflow

### PR DESCRIPTION
**Pull request by titodalcanton**
Configuration tuned for a bank equivalent to the O1 uberbank (slightly smaller, similar effectualness).

### CODE REVIEW

File: docs/resources/uberbank_example1.ini [hunk 1]
```diff
@@ -21,8 +21,14 @@ file-retention-level = merged_triggers
 
 ; Start and end times are needed to obey LIGO file naming conventions. Actual
 ; values are irrelevant. This can be overriden on the command line.
-start-time = 900000000
-end-time = 900010000
+; Here we use the times matching the estimated PSD so we have an immediate
+; reference on the resulting bank file name
+start-time = 1126051217
+end-time = 1137254417
+
```
Comment (spxiwh) [inline docs/resources/uberbank_example1.ini]: Please add a comment to explain that the options that follow are not really part of the [workflow] section, but are just stuck at the top and substituted in later. Maybe they could go in a separate section to emphasize that?

File: docs/resources/uberbank_example1.ini [hunk 2]
```diff
@@ -97,19 +102,19 @@ min-match = 0.97
...
-split-bank-num = 25
+min-mass1 = 1
+max-mass1 = 4
+min-mass2 = 1
+max-mass2 = 4
+min-total-mass = 2
+max-chirp-mass = 1.5
+max-bh-spin-mag = ${workflow|max-bh-spin}
+max-ns-spin-mag = ${workflow|max-ns-spin}
+ns-bh-boundary-mass = ${workflow|nsbh-boundary}
```
Comment (spxiwh) [inline docs/resources/uberbank_example1.ini]: Maybe stick the two PSD file inputs at the top as well, so that they live together. Then if a user changes one they will also clearly see that the other should be changed as well ... But we should probably persuade the codes to take the same format input one way or the other (or both)

File: docs/resources/uberbank_example1.ini [hunk 3]
```diff
@@ -134,46 +139,39 @@ approximant = SEOBNRv2_ROM_DoubleSpin_HI
...
-spin1-min = -0.9899
-spin2-max = 0.9899
-spin2-min = -0.9899
+mass1-min = 1
+mass1-max = 99
+mass2-min = 1
+mass2-max = 99
 
 [sbank-bbh-coarse]
 ; These options are sent only to the initial sbank-coarse job for "FINAL".
```
Comment (spxiwh) [inline docs/resources/uberbank_example1.ini]: Why this change?

File: docs/resources/uberbank_example1.ini [hunk 4]
```diff
@@ -134,46 +139,39 @@ approximant = SEOBNRv2_ROM_DoubleSpin_HI
...
+bh-spin-min = -${workflow|max-bh-spin}
+bh-spin-max = ${workflow|max-bh-spin}
+ns-spin-min = -${workflow|max-ns-spin}
+ns-spin-max = ${workflow|max-ns-spin}
+ns-bh-boundary-mass = ${workflow|nsbh-boundary}
+
 [sbank-bbh]
 ; These options are sent to all sbank "BBH" jobs
 
 ; Mass/spin parameter options
```
Comment (spxiwh) [inline docs/resources/uberbank_example1.ini]: So part of the idea of "BBH" and "final" was so that "BBH" would fill in the part of parameter space where signals have already been found and then final would fill everything else. "BBH" used a higher minimal match to ensure a dense coverage in the "BBH" region, with only a small increase in template count.

File: docs/resources/uberbank_example1.ini [hunk 5]
```diff
@@ -134,46 +139,39 @@ approximant = SEOBNRv2_ROM_DoubleSpin_HI
...
-mass2-max = 100.0
-mtotal-min = 10.0
-mtotal-max = 100.0
-mratio-min = 1.0
-mratio-max = 3.0
-aligned-spin =
-spin1-max = 0.9899
-spin1-min = -0.9899
-spin2-max = 0.9899
-spin2-min = -0.9899
```
Comment (spxiwh) [inline docs/resources/uberbank_example1.ini]: Just realised that I missed the ns-bh boundary mass in the original version. So the final bank would have populated BNS with large spins as well! That might explain a lot of the bank size increase.

File: docs/resources/uberbank_example1.ini (hunk 1 above)
Comment (titodalcanton) [inline docs/resources/uberbank_example1.ini]: Edit coming.

File: docs/resources/uberbank_example1.ini (hunk 2 above)
Comment (titodalcanton) [inline docs/resources/uberbank_example1.ini]: Edit coming.

File: docs/resources/uberbank_example1.ini (hunk 3 above)
Comment (titodalcanton) [inline docs/resources/uberbank_example1.ini]: I just removed options which were moved to the common section and made max masses a bit tighter since 100+1 is not part of the space.

File: docs/resources/uberbank_example1.ini (hunk 4 above)
Comment (titodalcanton) [inline docs/resources/uberbank_example1.ini]: That choice makes sense, however if you really want to keep the increase in template count small, I think you have to restrict the overcovered BBH region to m2 > 10 or something like that. Otherwise you also overcover part of the NSBH space, where you have many templates, and then the increase is no longer small. Let's discuss similar ideas on Wednesday's call maybe.

File: docs/resources/uberbank_example1.ini (hunk 5 above)
Comment (titodalcanton) [inline docs/resources/uberbank_example1.ini]: Actually my ubertest1 did not use your exact example file—I had already introduced the ns-bh boundary mass. I suspect the increase mostly came from choices of minimal match in sbank jobs (which maybe were near optimal in the pristine version of your example file, but not in my edited version) and from the fact that geom and BBH sbank had quite some overlap.

File: docs/resources/uberbank_example1.ini (hunk 1 above)
Comment (titodalcanton) [inline docs/resources/uberbank_example1.ini]: Hmm, the workflow base class actually does expect [workflow] to have those options:
https://github.com/ligo-cbc/pycbc/blob/master/pycbc/workflow/core.py#L551

**Comment by titodalcanton** [comment]
Landscape!!!! :(

File: docs/resources/uberbank_example1.ini (hunk 1 above)
Comment (spxiwh) [inline docs/resources/uberbank_example1.ini]: Sorry, I tend to put a comment *above* the line I'm commenting on (looks more natural in the "files changed" window). This is referring to lines 29 - 31 (max-bh-spin to nsbh-boundary)

File: docs/resources/uberbank_example1.ini (hunk 5 above)
Comment (spxiwh) [inline docs/resources/uberbank_example1.ini]: Okay let's discuss this and the point above tomorrow. My suspicion is that most of the additional templates come from having all NSs (BNS *and* NSBH) with spin up to 0.99 in the original version .... that's a huge amount of parameter space, but I'm running some tests to see. I'll also think about the double coverage.

File: docs/resources/uberbank_example1.ini (hunk 3 above)
Comment (spxiwh) [inline docs/resources/uberbank_example1.ini]: Here I'm referring to line 174/176 (changing match-min to 0.9 for the sbank-bbh-coarse job).

Comment (spxiwh) [inline docs/resources/uberbank_example1.ini]: Actually that would probably be needed given that your "bbh" region is much larger than mine. I guess this choice depends on how that change goes.

**Comment by spxiwh** [comment]
Yeah, landscape does not automatically rebase, so can sometimes report erroneously if branches are a little out of date. Let's resolve the changes first and then we just ignore landscape as this is only .ini file changes, and landscape does not check .ini files anyway.

**Comment by titodalcanton** [comment]
Ian, after our telecon, would you prefer to stick to your original .ini file in the documentation (corrected for NS spin bounds etc)? It may still be good to have a .ini file which makes an O1-equivalent uberbank though, for reference.

**Comment by spxiwh** [comment]
Hi Tito, I think *most* of the changes here should be applied. However, I would like to revert the changes to the 'bbh' region of the template bank, keeping this (as in the original .ini file) covering from mtotal 10-100 with a mass ratio cut of 3 *and* with a minimal match of 0.98.

I would also prefer to keep the original geometric lattice boundary, but there is no real difference between the original setting and the one you are proposing.

Other than that I am happy with the changes. If you can update for that, this should be ready for merging.

**Comment by titodalcanton** [comment]
Hi Ian, I just reverted the sbank BBH and geom boundary settings. I'm going to do one more test with the new config to make sure I didn't break anything before merging.

**Comment by spxiwh** [comment]
Thanks Tito. All of this looks good to me now. I'm happy to merge this once your test is finished.

**Comment by titodalcanton** [comment]
I added a section to the template bank page with the test:
https://www.lsc-group.phys.uwm.edu/ligovirgo/cbcnote/PyCBC/O2TemplateBank#A.2BAH4-tito.2BAC8-o2.2BAC8-bank.2BAC8-ubertest4ian_.40_Atlas
The distribution of points looks quite "patchy". I doubt there are coverage issues, but maybe we should check... In the mean time we can probably merge this PR though.

**Comment by spxiwh** [comment]
I expected some of those patches, but this is a little weird and I'm not sure I understand all of the patches. Merging this, but we should probably try to understand this better!
